app/models/paper.rb

```diff
@@ -66,9 +67,4 @@ def autosave_associated_records_for_authors
     end
   end
 
-  def autosave_associated_records_for_links
-    self.links = links.map do |link|
-      Link.find_or_create_by name: link.name
-    end
-  end
```

Comment:
here's my solution.  I think we can get rid of this since we are checking for existing links for the link locator type.  I added a validation to the model so it currently only allows a link once for a paper.  We can make the validation unique by URL, but I figure this will probably be an edge case that we can always fix in postgres later.

I'm a little more concerned that a uniqueness validation for URL could break on an import if the URLs are not unique in crossref.